drivers/gpu/drm/sun4i/sun4i_hdmi_ddc_clk.c

```diff
@@ -109,38 +116,34 @@ static const struct clk_ops sun4i_ddc_ops = {
 	.set_rate	= sun4i_ddc_set_rate,
 };
 
-int sun4i_ddc_create(struct sun4i_hdmi *hdmi, struct clk *parent)
+struct clk *sun4i_ddc_create(struct device *dev, struct regmap *regmap,
+			     const struct sun4i_hdmi_i2c_variant *variant,
+			     const struct clk *parent_clk)
 {
 	struct clk_init_data init;
 	struct sun4i_ddc *ddc;
-	const char *parent_name;
+	const char *parents[SUN4I_HDMI_I2C_CLK_NUM_PARENTS];
```

Comment:
This creates a local (on stack) array of 1 character (define is 1), yet, this is being passed on indirectly as a pointer to other places using structures. See comment further on.